### PR TITLE
JAMES-3925 - JMAP Upload - cleanup/fixup/optimize upload usage 

### PR DESCRIPTION
**Pull request by vttranlina**

### CODE REVIEW

**Review by hungphan227 — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

File: server/data/data-jmap/src/test/java/org/apache/james/jmap/api/upload/UploadRepositoryContract.scala [hunk 1]
```diff
@@ -188,4 +188,16 @@
        .isNotNull
    }
 
+   @Test
+   def deleteShouldReturnTrueWhenRowExists(): Unit = {
+     val uploadId: UploadId = SMono.fromPublisher(testee.upload(data(), CONTENT_TYPE, USER)).block().uploadId
+
+     assertThat(SMono.fromPublisher(testee.delete(uploadId, USER)).block()).isTrue
+   }
+   @Test
```
Comment (chibenwa) [inline server/data/data-jmap/src/test/java/org/apache/james/jmap/api/upload/UploadRepositoryContract.scala:197]: Missing LF

File: server/data/data-jmap-cassandra/src/main/java/org/apache/james/jmap/cassandra/upload/UploadDAO.java [hunk 2]
```diff
@@ -171,6 +171,7 @@ public UploadDAO(CqlSession session, BlobId.Factory blobIdFactory) {
         this.delete = session.prepare(deleteFrom(TABLE_NAME)
             .whereColumn(USER).isEqualTo(bindMarker(USER))
             .whereColumn(ID).isEqualTo(bindMarker(ID))
+            .ifExists()
```
Comment (chibenwa) [inline server/data/data-jmap-cassandra/src/main/java/org/apache/james/jmap/cassandra/upload/UploadDAO.java:174]: Using a LWT? Are we sure we need strong consistency?

**Review by chibenwa — CHANGES_REQUESTED** [review]
Using LWTs is a nogo for me.

**Review by chibenwa — APPROVED** [review]

**Comment by Arsnael** [comment]
Can you squash the fixup @vttranlina ?

**Comment by vttranlina** [comment]
squash fixup

**Pull request merged**